src/core/profile-provider/resolve-map-ast.ts

```diff
@@ -0,0 +1,116 @@
+import type {
+  MapDocumentNode,
+  NormalizedSuperJsonDocument,
+} from '@superfaceai/ast';
+import { assertMapDocumentNode, EXTENSIONS } from '@superfaceai/ast';
+
+import type { IConfig, IFileSystem, ILogger } from '../../interfaces';
+import {
+  profileIdsDoNotMatchError,
+  profileNotFoundError,
+  profileProviderNotFoundError,
+  providersDoNotMatchError,
+  referencedFileNotFoundError,
+  sourceFileExtensionFoundError,
+  unsupportedFileExtensionError,
+  variantMismatchError,
+} from '../errors';
+
+const DEBUG_NAMESPACE = 'map-file-resolution';
+
+export async function resolveMapAst({
+  profileId,
+  providerName,
+  variant,
+  superJson,
+  fileSystem,
+  logger,
+  config,
+}: {
+  profileId: string;
+  providerName: string;
+  variant?: string;
+  logger?: ILogger;
+  superJson: NormalizedSuperJsonDocument | undefined;
+  fileSystem: IFileSystem;
+  config: IConfig;
+}): Promise<MapDocumentNode | undefined> {
+  if (superJson === undefined) {
+    return undefined;
+  }
+  const profileSettings = superJson.profiles[profileId];
+
+  if (profileSettings === undefined) {
+    throw profileNotFoundError(profileId);
+  }
+
+  const profileProviderSettings = profileSettings.providers[providerName];
+
+  if (profileProviderSettings === undefined) {
+    throw profileProviderNotFoundError(profileId, providerName);
+  }
+
+  const log = logger?.log(DEBUG_NAMESPACE);
+  if ('file' in profileProviderSettings) {
+    let path: string;
+    if (profileProviderSettings.file.endsWith(EXTENSIONS.map.source)) {
+      path = fileSystem.path.resolve(
+        fileSystem.path.dirname(config.superfacePath),
+        profileProviderSettings.file.replace(
+          EXTENSIONS.map.source,
+          EXTENSIONS.map.build
+        )
+      );
+      // check if ast exists to print usefull error (needs to be compiled)
+      if (!(await fileSystem.exists(path))) {
+        throw sourceFileExtensionFoundError(EXTENSIONS.map.source);
+      }
```

Comment:
question: Is this necessary when you're checking bellow whether reading file failed or not?

Reply:
This is there to throw correct error when source file is not compiled.